noir-projects/aztec-nr/aztec/src/discovery/private_logs.nr

```diff
@@ -31,14 +34,32 @@ use crate::encrypted_logs::log_encryption::PRIVATE_LOG_PLAINTEXT_SIZE_IN_FIELDS;
 /// are stored in the PXE capsules so that `fetch_and_process_public_partial_note_completion_logs` can later search for
 /// public logs that will complete them.
 pub unconstrained fn fetch_and_process_private_tagged_logs<Env>(
-    _contract_address: AztecAddress,
-    _compute_note_hash_and_nullifier: ComputeNoteHashAndNullifier<Env>,
+    contract_address: AztecAddress,
+    compute_note_hash_and_nullifier: ComputeNoteHashAndNullifier<Env>,
 ) {
     // We will eventually fetch tagged logs, decrypt and process them here, but for now we simply call the `syncNotes`
-    // oracle. This has PXE perform tag synchronization, log download, decryption, and finally calls to the the
-    // `process_log` contract function with the decrypted payload, which will in turn call `do_process_log` with a
-    // decrypted log, letting us continue the work outside of PXE.
+    // oracle. This makes PXE synchronize tags, download logs and store the logs in capsules which are then retrieved
+    // and processed here.
     sync_notes();
+
+    // Get logs from capsules
+    let logs = CapsuleArray::<LogCapsule>::at(contract_address, LOG_CAPSULE_ARRAY_BASE_SLOT);
+    let len = logs.len();
+
+    for i in 0..len {
+        let log = logs.get(i);
+
+        // Process each log
+        process_log(contract_address, compute_note_hash_and_nullifier, log);
+    }
+
+    // TODO: Introduce array.pop() or array.wipe()?
```

Comment:
What does the reviewer think? I think it would make sense to introduce pop method but wanted to get feedback before implementing that.

Reply:
https://github.com/AztecProtocol/aztec-packages/pull/13107/commits/55a499155a912b8bfef64106a3edea90adcd6882